diff --git a/src/Plugin/GraphQL/DataProducer/FetchEntity.php b/src/Plugin/GraphQL/DataProducer/FetchEntity.php
index ffef6a8..7c84154 100644
--- a/src/Plugin/GraphQL/DataProducer/FetchEntity.php
+++ b/src/Plugin/GraphQL/DataProducer/FetchEntity.php
@@ -250,6 +250,35 @@ class FetchEntity extends DataProducerPluginBase implements ContainerFactoryPlug
         }
       }
 
+      // Autosave: get autosaved values.
+      if (\Drupal::service('module_handler')->moduleExists('silverback_autosave')) {
+        $context->mergeCacheMaxAge(0);
+        // @todo Add DI to both.
+        $service = \Drupal::service('silverback_autosave.entity_form_storage');
+        /**
+         * Quick and dirty because this causes leaked metadata error.
+         * $entityForm = \Drupal::service('entity.form_builder')->getForm($entity, 'edit');
+         * $form_id = $entityForm['form_id']['#value'];
+         */
+         $form_id = "{$entity->getEntityTypeId()}_{$entity->bundle()}_edit_form";
+         $autosaved_state = $service->getEntityAndFormState($form_id, $entity->getEntityTypeId(), $entity->id(), $entity->language()->getId(), \Drupal::currentUser()->id());
+         /** @var \Drupal\Core\Entity\EntityInterface $autosaved_entity */
+        $autosaved_entity = $autosaved_state['entity'] ?? NULL;
+        /** @var \Drupal\Core\Form\FormStateInterface $autosaved_form_state */
+        $autosaved_form_state = $autosaved_state['form_state'] ?? [];
+        if ($autosaved_entity && !empty($autosaved_form_state)) {
+          $current_user_input = $autosaved_form_state->getUserInput();
+          foreach ($autosaved_entity->getFields() as $name => $field) {
+            if (in_array($name, ['title', 'body']) || str_starts_with($name, 'field_')) {
+              if (isset($current_user_input[$name])) {
+                $field->setValue($current_user_input[$name]);
+              }
+            }
+          }
+          return $autosaved_entity;
+        }
+      }
+
       return $entity;
     });
   }
